scripts/certs.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+set -eou pipefail
+
+NAMESPACE="openebs"
+APP_NAME="api-rest"
+CERT_SECRET_NAME="api-rest-tls"
+CERT_DIR="$(dirname "$0")/certs"
+
+rm -rf "${CERT_DIR}"
+mkdir -p "${CERT_DIR}"
+
+# Create a self-signed root CA
+echo "Creating a self-signed root CA"
+openssl genrsa -out "${CERT_DIR}/ca.key" 4096
+openssl req -x509 -new -nodes -key "${CERT_DIR}/ca.key" -sha256 -days 3650 -out "${CERT_DIR}/ca.crt" -subj "/CN=api-rest-ca" -addext "subjectAltName=DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc.cluster.local,DNS:${NAMESPACE}-${APP_NAME},DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc"
+
+# Create TLS certificate for the API REST
+echo "Creating a TLS certificate for the API REST"
+openssl genrsa -out "${CERT_DIR}/server.key" 4096
+openssl req -new -key "${CERT_DIR}/server.key" -out "${CERT_DIR}/server.csr" -subj "/CN=${NAMESPACE}-${APP_NAME}" -addext "subjectAltName=DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc.cluster.local,DNS:${NAMESPACE}-${APP_NAME},DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc"
+openssl x509 -req -in "${CERT_DIR}/server.csr" -CA "${CERT_DIR}/ca.crt" -CAkey "${CERT_DIR}/ca.key" -CAcreateserial -out "${CERT_DIR}/server.crt" -days 3650 -sha256 -extfile <(printf "subjectAltName=DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc.cluster.local,DNS:${NAMESPACE}-${APP_NAME},DNS:${NAMESPACE}-${APP_NAME}-${NAMESPACE}.svc")
+
+# Convert the private key to PKCS#1 format if necessary
+echo "Verifying the RSA key format"
+if grep -q "BEGIN PRIVATE KEY" "${CERT_DIR}/server.key"; then
+  echo "Converting key to RSA format"
+  openssl rsa -in "${CERT_DIR}/server.key" -out "${CERT_DIR}/server-rsa.key"
+  mv "${CERT_DIR}/server-rsa.key" "${CERT_DIR}/server.key"
+else
+  echo "Key is already in RSA format"
+fi
+
+# Create a Kubernetes secret
+echo "Creating a Kubernetes secret"
+kubectl create secret generic ${CERT_SECRET_NAME} \
```

Comment:
Should we delete and recreate if `${CERT_SECRET_NAME}` exists?

Reply:
Add a blank line at the end?

Reply:
will do,  I created this cert file for testing purposes only until we get cert-manager installed... The plan is to not rely on this script and to use cert-manager to create all the certs.  I am open to discuss and direction on this.